WELL-ARCHITECTED.md

```diff
@@ -0,0 +1,210 @@
+>
+>* We do not have a defined KPI within the application. We could however use a metric such as number of records processed within a given time frame and alert if this is outside of the defined thresholds.
+
+---
+
+#### OPS 2. How do you approach application lifecycle management?
+
+* [ ] Question does not apply to this workload
+
+* [x] **[Required]** Use infrastructure as code and stages isolated in separate environments
+* [x] **[Good]** Prototype new features using temporary environments
+* [ ] **[Good]** Use a rollout deployment mechanism
+* [ ] **[Good]** Use configuration management
+* [ ] **[Good]** Review the function runtime deprecation policy
+* [ ] **[Best]** Use CI/CD including automated testing across separate accounts
+
+* [ ] None of these
+
+
+##### Notes
+
+>* Our example utilizes infrastructure as code and includes a simple pipeline that will build and deploy within an individual account and to an individual environment. However the nature of this example means it can be deployed many times it's possible. You can for example deploy a staging pipeline that would watch a development branch and deploy and changes to the Staging application stack. You could also deploy a production pipeline stack that watches the master branch and merges here will trigger a production release.
+>
+>* For this example a rollout mechanism would involve adopting either a Blue / Green deployment strategy with you controlling which input bucket a particular user hits . Alternatively for application business logic only changes these could be tested by having a notification invoke an alternate version of a lambda under specific conditions. 
+
+---
+
+## Security
+
+#### SEC 1: How do you control access to your Serverless API?
+
+* [x] Question does not apply to this workload
+
+
+* [ ] **[Required]** Use appropriate endpoint type and mechanisms to secure access to your API
+* [ ] **[Good]** Use authentication and authorization mechanisms
+* [ ] **[Best]** Scope access based on identity’s metadata
+
+
+* [ ] None of these
+
+
+##### Notes
+
+>This solution doesn't include an API frontend so the question doesn't apply.
+
+---
+
+#### SEC 2: How do you manage your Serverless application’s security boundaries?
+
+* [ ] Question does not apply to this workload
+
+
+* [x] **[Required]** Evaluate and define resource policies
+* [x] **[Good]** Control network traffic at all layers
+* [x] **[Best]** Smaller functions require fewer permissions
+* [x] **[Required]** Use temporary credentials between resources and components
+
+* [ ] None of these
+
+
+##### Notes
+
+> * We use IAM policy to ensure that resources can only be called by other resources that should be calling them.
+>
+> * The pipeline will assume a role with only the permissions it requires to deploy the application. This will either be only being able to perform a specific action on multiple resources or any action on a particular resource. 
```

Comment:
I think it better we call out the application in general, not the pipeline.

Reply:
I updated the wording to speak in application wide terms.